specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,649 @@
+            })
+            .Get(),
+        &m_contextMenuRequestedToken);
+```
+
+## Win32 C++ Use Data to Display Custom Context Menu
+
+The developer can use the data provided in the Event arguments to display a custom context menu with entries of their choice. For this case, the developer specifies Handled to be true and requests a deferral. Deferral of this event should be completed when the user selects a context menu item (either the app developer will handle the case, or can return the selected option to the WebView) or when they click on the screen (effectively closing the menu).
+
+ ```cpp
+    webview2->add_ContextMenuRequested(
+        Callback<ICoreWebView2ContextMenuRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2ContextMenuRequestedEventArgs* args)
+            {
+                auto showMenu = [this, args]
+                {
+                    wil::com_ptr<ICoreWebView2ContextMenuItemCollection> items;
+                    CHECK_FAILURE(args->get_MenuItems(&items));
+                    args->put_Handled(true);
+
+                    UINT32 itemsCount;
+                    CHECK_FAILURE(items->get_Count(&itemsCount));
+
+                    HMENU hPopupMenu = CreatePopupMenu();
+                    wil::com_ptr<ICoreWebView2ContextMenuItem> current;
+
+                    for (UINT32 i = 0; i < itemsCount; i++) 
+                    {
+                        CHECK_FAILURE(items->GetValueAtIndex(i, &current));
+                        wil::unique_cotaskmem_string name;
+                        CHECK_FAILURE(current->get_Name(&name));
+                        UINT32 command_id;
+                        CHECK_FAILURE(current->get_CommandId(&command_id));
+                        InsertMenu(hPopupMenu, 0, MF_BYPOSITION | MF_STRING, command_id, name);
+                    }
+                    HWND hWnd;
+                    m_appWindow->GetWebViewController()->get_ParentWindow(&hWnd);
+                    SetForegroundWindow(hWnd);
+                    wil::com_ptr<ICoreWebView2ContextMenuInfo> parameters;
+                    args->get_ContextMenuInfo(&parameters);
+                    POINT locationInControlCoordinates;
+                    POINT locationInScreenCoordinates;
+                    CHECK_FAILURE(parameters->get_Location(&locationInControlCoordinates));
+                    // get_Location returns coordinates in relation to upper left Bounds of the WebView2.Controller. Will need to convert to Screen coordinates to display the popup menu in the correct location.
+                    ConvertToScreenCoordinates(locationInControlCoordinates, &locationInScreenCoordinates);
+                    UINT32 selectedCommand = TrackPopupMenu(hPopupMenu, TPM_TOPALIGN | TPM_LEFTALIGN | TPM_RETURNCMD, locationInScreenCoordinates.x, locationInScreenCoordinates.y, 0, hWnd, NULL);
+                    CHECK_FAILURE(args->put_SelectedCommand(selectedCommand));
```

Comment:
The reason I have them return the unique command ID here is because I thought it would be easier for the developer to return an id that was given to them rather than having to return the selected ContextMenuItem object, especially if the object selected is nested inside a few layers of submenus (would be hard to retrieve an item based off index when submenus come into play). If you feel like that is something the developer will have to work around, let me know and I can change it.